core/server/master/src/main/java/alluxio/master/file/meta/TtlBucketList.java

```diff
@@ -145,4 +165,39 @@ public void remove(InodeView inode) {
   public void removeBuckets(Set<TtlBucket> buckets) {
     mBucketList.removeAll(buckets);
   }
+
+  @Override
+  public CheckpointName getCheckpointName() {
+    return CheckpointName.TTL_BUCKET_LIST;
+  }
+
+  @Override
+  public void writeToCheckpoint(OutputStream output) throws IOException, InterruptedException {
+    CheckpointOutputStream cos = new CheckpointOutputStream(output, CheckpointType.LONGS);
+    for (TtlBucket bucket : mBucketList) {
+      for (Inode inode : bucket.getInodes()) {
+        cos.writeLong(inode.getId());
+      }
+    }
+  }
+
+  @Override
+  public void restoreFromCheckpoint(CheckpointInputStream input) throws IOException {
+    mBucketList.clear();
+    Preconditions.checkState(input.getType() == CheckpointType.LONGS,
+        "Unexpected checkpoint type: %s", input.getType());
+    while (true) {
+      try {
+        long id = input.readLong();
+        Optional<Inode> inode = mInodeStore.get(id);
```

Comment:
Does this imply there is a required order for the checkpointing? Is that mentioned somewhere?